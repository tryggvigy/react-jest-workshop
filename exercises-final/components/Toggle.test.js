import React from 'react';
import { shallow } from 'enzyme';
import Toggle from './Toggle';

test('has toggle--off class applied by default', () => {
  const wrapper = shallow(<Toggle onToggle={() => {}} children="Toggle Me" />);
  expect(wrapper.hasClass('toggle--off')).toBe(true);
});

test('has toggle--on class applied when initialToggledOn specified to true', () => {
  const wrapper = shallow(<Toggle onToggle={() => {}} children="Toggle Me" initialToggledOn />);
  expect(wrapper.hasClass('toggle--on')).toBe(true);
});

test('invokes the onToggle prop when clicked', () => {
  const onToggle = jest.fn();
  const wrapper = shallow(<Toggle children="Toggle Me" onToggle={onToggle} />);
  wrapper
    .find('button')
    .first()
    .prop('onClick')();
  expect(onToggle).toHaveBeenCalledTimes(1);
  expect(onToggle).toBeCalledWith(true);
});
