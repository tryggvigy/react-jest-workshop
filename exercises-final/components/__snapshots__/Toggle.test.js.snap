// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="toggle toggle--off"
>
  <button
    onClick={[Function]}
  >
    Toggle Me
  </button>
</div>
`;
